lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -610,31 +646,80 @@ private function getAutoincrementIdentifierName(Identifier $table)
      */
     public function getListTableForeignKeysSQL($table)
     {
-        $table = $this->normalizeIdentifier($table);
-        $table = $this->quoteStringLiteral($table->getName());
+        return $this->getListForeignKeysSQL($table);
+    }
+
+    /**
+     * Returns the list of foreign keys for the current database.
+     *
+     * @param string $table
+     * @param string $database
+     *
+     * @return string
+     */
+    public function getListForeignKeysSQL($table = null, $database = null)
```

Comment:
There should be two (sets of) methods: `getTableSomething($table)` and `getAllSomething()`. Otherwise, the single function has two paths which don't share much, and a mistakenly defined value of `$table` may change the resulting function logic.